### ISSUE TITLE
ES stays as a background process when RetroArch 1.2 is closed

### ISSUE THREAD

**Issue by Mr-Figs** [Aloshi/EmulationStation#474]
With the new Retroarch release (1.2 stable, Windows x64), ES can still launch games like normal, but on my setup, once a game is closed by quitting RetroArch, ES does not reopen its window as it did before. Instead, ES remains as a background process, like it normally does when a game is being run. Multiple instances can be run at once if they are not manually terminated. Alt+F4 and the menu option to quit still work fine. `es_log.txt` is as follows after a game is launched (the log is not changed by the ending of the process, of course).
```
lvl2: 	EmulationStation - v2.0.1a, built Mar  8 2015 - 16:10:39
lvl2: 	Creating surface...
lvl2: 	Created window successfully.
lvl2: 	Added known joystick XInput Controller #1 (instance ID: 0, device index: 0)
lvl2: 	Checking available OpenGL extensions...
lvl2: 	 ARB_texture_non_power_of_two: ok
lvl2: 	Loading system config file C:/Users/Lorenzo/.emulationstation/es_systems.cfg...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/nes/gamelist.xml"...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/snes/gamelist.xml"...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/megadrive/gamelist.xml"...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/n64/gamelist.xml"...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/psx/gamelist.xml"...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/gba/gamelist.xml"...
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/nds/gamelist.xml"...
lvl1: 	  Unknown platform for system "mame" (platform "mame" from list "mame")
lvl2: 	Parsing XML file "C:/Users/Lorenzo/.emulationstation/gamelists/mame/gamelist.xml"...
lvl2: 	 req sound [detailed.launch]
lvl2: 	   (missing)
```

RetroArch is the clear culprit, but it might be possible to fix it on ES' end—but that depends on how this is happening. Any theories?

Also, what does `(missing)` mean? I can't find anything on it.

**Comment by activegen** [comment]
I experience the same issue on my system, both with the new version of Retroarch (1.2.2) and the previous. 

Could this have something to do with a Windows update? In my case the problem just appeared. One day it worked fine, the next it didn't. 

My system specs are Windows 8.1 x64 & the latest version of Emulation Station.

**Comment by RJRetro** [comment]
Having this issue myself.  It worked fine before updating to Windows 10 x64.  Before that, I was using Windows 7 x64 on one machine and 8.1 x64 on another.  Using Retroarch 1.2.2.  After exiting a game, all I see is the command window.

**Comment by RJRetro** [comment]
Ok update here... it actually is working on one of my machines, the one that updated from Windows 7x64 but not the other one.  My mistake.  However, I've discovered that Retroarch isn't to blame.  I launch MAME from EmulationStation as well and it does the same thing.  I thought maybe it had to do with me using the new Windows 10 command prompt (turning off legacy mode) but I changed it back and rebooted to no effect.

**Comment by navnakotnes** [comment]
same problem here. tested with win 7 x64 ult edition on two separate machines.
Only important microsoft updates are installed on both of them
one of them has intel graphics and doesn't have this problem
the other one has amd and keeps having this problem (was ok 2 weeks ago)

**Comment by RJRetro** [comment]
My culprit machine is AMD also.  I performed a clean install of Windows 10, but the problem didn't go away.  The PC I have that it works fine with is an nVidia.

**Comment by navnakotnes** [comment]
I think the issue might have appeared when I updated my machine to the newest amd drivers.
im gonna stick with 2.0.0-rc1 for now
maybe an older version of amd drivers could fix this

**Comment by RJRetro** [comment]
I'm now a hundred percent sure this is something to do with an AMD driver.  I installed an older driver and it fixed this issue.  This killed my ability to use Catalyst Control Center however and also caused ES to have really bad flickering, so I let it update back to the newer driver and now once again, ES won't relaunch when exiting a game.

**Comment by Mr-Figs** [comment]
I have an AMD card as well, so that checks out. I've also done some slightly-better debugging; the command `emulationstation --debug --resolution 1280 720 --windowed --ignore-gamelist` returns the output
```
lvl2:   EmulationStation - v2.0.1a, built Mar  8 2015 - 16:10:39
lvl2:   Creating surface...
lvl2:   Created window successfully.
lvl2:   Added known joystick XInput Controller #1 (instance ID: 0, device index: 0)
lvl2:   Checking available OpenGL extensions...
lvl2:    ARB_texture_non_power_of_two: ok
lvl2:   Loading system config file C:/Users/Lorenzo/.emulationstation/es_systems.cfg
...
lvl1:     Unknown platform for system "mame" (platform "mame" from list "mame")
lvl2:    req sound [basic.launch]
lvl2:      (missing)
lvl2:   Attempting to launch game...
lvl2:    getting volume as 20 ( from float 0.2)
lvl2:           ESOpen.bat "C:\Program Files\Libretro\cores\nestopia_libretro.dll" "
E:\roms\nes\Castlevania.nes""
==============================================
==============================================
lvl2: 	Creating surface...
```
So if the "Creating surface" line has to do with the AMD drivers, we have a lead. The Microsoft tablet by the same name is making research difficult however.

By the way, I use a batch file (ESOpen.bat) to launch games—it was a troubleshooting step, but it didn't change anything (games still launch, but ES stays hidden).

**Comment by RJRetro** [comment]
I'm reading that people didn't have this issue with a previous version of ES.  What changed,  I wonder?

**Comment by Aloshi** [comment]
It could be that I changed the "launch game" function to use the Unicode-supporting version `_wsystem()` in the latest release, and Microsoft changed how it worked in an update.

See [this function](https://github.com/Aloshi/EmulationStation/blob/master/es-core/src/platform.cpp#L60). If anyone here knows how to compile ES, you might want to removing the WIN32 #ifdef, changing the entire block to be:

```cpp
int runSystemCommand(const std::string& cmd_utf8)
{
	return system(cmd_utf8.c_str());
}
```
This should match the old behavior (though it's been quite a while since I looked at the code).

**Comment by navnakotnes** [comment]
amd 15.8 Beta drivers fixes this

**Comment by Mr-Figs** [comment]
Just downloaded them, and yes they do. The change log just mentions specific games and windows features though, so it's hard to tell just what AMD changed back to normal. But as long as it works…right?

**Comment by RJRetro** [comment]
amd beta drivers aren't an option for me.  Catalyst installer won't show the driver and I've tried all the tricks.

**Comment by navnakotnes** [comment]
updating from amd catalyst control center doesn't work for some reason
did you try downloading them from amd's site
here http://support.amd.com/en-us/kb-articles/Pages/latest-catalyst-windows-beta.aspx

**Comment by Mr-Figs** [comment]
That's what I had to do as well. Even after installing the beta drivers, 
Catalyst says that the beta isn't installed under Software Update, but 
the Software page shows that it is running a newer version. Sounds like 
an AMD issue though.

------ Original Message ------

**Comment by RJRetro** [comment]
Yeah I went through all of that yesterday and then some with my laptop and its Radeon HD 6520G, which I think might not be compatible.  As Mr-Figs said tho, that's on the AMD end so if anybody can offer assistance, I've made my e-mail public on my profile.  Meantime, I've switched back to ES 2.0.0 RC1.

**Comment by Tetriser** [comment]
Me too... Reverted to 2.0.0 rc-1

Windows 8.1 x64, AMD Radeon R5

**Comment by RJRetro** [comment]
Finally, I have some luck!  Tetriser, try installing the newest beta.  It's 15.9.something now, it installed on my machine fine, and latest ES works flawless again :-D

**Comment by Tetriser** [comment]
It seems to return ES back to desktop on after upgrading Windows 10.

@RJRetro - Sorry for the late response, I don't know that a new version is released! Not visiting GitHub for months.
